### DISCUSSION TITLE
FR: "Sources with URLs" option for an Element field's source option

### DISCUSSION THREAD

**Discussion by timkelty** [craftcms/cms#8555] [Ideas]
It would be great if element selection fields (Elements, Categories, etc) had a dynamic option (similar to **All**) that showed "Sources with URLs".

Example use case:

An Entries field, "Link To…", which determines the target of a link. You want the user to be able to select any element with a URL, but don't want to have to edit the field every time a new section/category group/etc with URLs are added.

**Comment by davist11** [comment]
Yes please! Was just going to add a FR for this.

**Comment by mmikkel** [comment]
A sorely missed feature – especially since Craft still has no native LinkIt-ish fieldtype, which makes link creation a very common use case for Entries fields. Gets quite tiresome to manually check all the sections with URLs in the field settings.
